src/main/java/si/mazi/rescu/clients/ApacheConnection.java

```diff
@@ -0,0 +1,217 @@
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.InetSocketAddress;
+import java.net.MalformedURLException;
+import java.net.ProtocolException;
+import java.net.Proxy;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLSocketFactory;
+
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpHost;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpDelete;
+import org.apache.http.client.methods.HttpEntityEnclosingRequestBase;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpHead;
+import org.apache.http.client.methods.HttpOptions;
+import org.apache.http.client.methods.HttpPatch;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpPut;
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.entity.ByteArrayEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.HttpClients;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import si.mazi.rescu.HttpMethod;
+
+public class ApacheConnection implements HttpConnection {
+    
+    private static final Logger log = LoggerFactory.getLogger(ApacheConnection.class);
+    
+    private final String url;
+    private final Proxy proxy;
+    private final HttpClientBuilder builder = HttpClients.custom();
+    private final RequestConfig.Builder requestConfig = RequestConfig.custom();
+            
+    private boolean executed = false;
+    private HttpMethod method = HttpMethod.GET;
+    private ByteArrayOutputStream out;
+    private final Map<String, String> headers = new HashMap<>();
+    
+    private CloseableHttpResponse response;
+
+    private ApacheConnection(String url, Proxy proxy) {
+        super();
+        this.url = url;
+        this.proxy = proxy;
+    }
+    
+    public static HttpConnection create(String url, Proxy proxy) throws MalformedURLException, IOException {
+        return new ApacheConnection(url, proxy);
+    }
+    
+    private void exec() throws ClientProtocolException, IOException {
+        if (executed) {
+            return;
+        }
+        HttpClientBuilder clientBuilder = builder
+                .setDefaultRequestConfig(requestConfig.build());
+        if (proxy != null && proxy != Proxy.NO_PROXY) {
+            InetSocketAddress address = (InetSocketAddress) proxy.address(); 
+            HttpHost proxy = new HttpHost(address.getHostName(), address.getPort());
+            clientBuilder.setProxy(proxy);
+        }
+        try ( CloseableHttpClient client = clientBuilder.build() ) {
```

Comment:
This try-with-resources will close the connection, and the this.response will be closed, after that, getInputStream() will not work as the connection and the stream are both closed. And result in this exception:
org.apache.http.ConnectionClosedException: Premature end of chunk coded message body: closing chunk expected

Reply:
without going too deep into this, just one question:
have you tried it und received a ConnectionClosedException ?
using the implementation for quite some time and never seen this one.
could you provide test code to reproduce ?

Reply:
When the response is chunked, the `org.apache.http.ConnectionClosedException: Premature end of chunk coded message body: closing chunk expected` will happen:

```
2025-03-11 23:34:04,820 TRACE  [main] s.m.r.HttpTemplate.receive:162 Header response property: key='Transfer-Encoding', value='[chunked]'
```

Reply:
ok, i see the issue in theory, i mean without having reproduced it locally...
@sutra any suggestions? so, better not use the 'try-with-resources' so, the client is not closed?
feel free to make a PR

Reply:
A demo to consume the chunked response: https://github.com/sutra/webflux-demo

mvn spring-boot:run
and then run the ApacheConnectionTest

you will see the `testApacheConnectionChunked` failed.

Reply:
again, thank you for the catch, and i see the problem
> any suggestions? so, better not use the 'try-with-resources' so, the client is not closed?
> feel free to make a PR

Reply:
Created a PR #152